test/cases/base/load_test.rb

```diff
@@ -91,6 +91,17 @@ def setup
     @person = Person.new
   end
 
+  def test_writing_attribute_retains_value_instance
+    value = { not: "an ARes instance" }
+    person = Person.new
+    person.non_ar_hash = value
+    person.non_ar_hash[:not] = "changed"
+
+    skip "TODO: Failing due to indifferent access"
```

Comment:
This will hopefully pass once the `Hash#with_indifferent_access` call is removed from `#attributes`.